### PR TITLE
[ZTW-983] Added SNS and SQS permissions for tags

### PR DESCRIPTION
**Pull request by jrguzik**
Modification to Zscaler terraform IAM module to support SMEdge creating CreateQueue and Subscribe for internal tags feature.

### CODE REVIEW

File: modules/terraform-zscc-iam-aws/main.tf [hunk 1]
```diff
@@ -120,6 +120,36 @@ resource "aws_iam_role_policy_attachment" "cc_session_manager_attachment" {
...
+    actions = [
+      "sqs:CreateQueue",
+      "sns:Subscribe",
+    ]
+    resources = ["*"]
+  }
+}
+
+resource "aws_iam_policy" "cc_tags_policy" {
+  count       = var.byo_iam == false ? var.iam_count : 0
```
Comment (eliassun) [inline modules/terraform-zscc-iam-aws/main.tf:140]: It needs a flag to turn on/off. The workload tagging feature is not a must.

File: modules/terraform-zscc-iam-aws/main.tf [hunk 2]
```diff
@@ -120,6 +120,36 @@ resource "aws_iam_role_policy_attachment" "cc_session_manager_attachment" {
...
+  description = "Policy which permits CCs to subscribe for tags changes"
+  name        = "${var.name_prefix}-cc-${count.index + 1}-tags-${var.resource_tag}"
+  policy      = data.aws_iam_policy_document.cc_tags_policy_document.json
+}
+
+resource "aws_iam_role_policy_attachment" "cc_tags_attachment" {
+  count      = var.byo_iam == false ? var.iam_count : 0
```
Comment (eliassun) [inline modules/terraform-zscc-iam-aws/main.tf:147]: It needs a flag to turn on/off. The workload tagging feature is not a must.

**Review by eliassun — CHANGES_REQUESTED** [review]
We need a flag to turn/off these resource. The workload tagging is not a must.

**Comment by jrguzik** [comment]
Done - default is false

File: modules/terraform-zscc-iam-aws/main.tf (hunk 2 above)
Comment (jrguzik) [reply modules/terraform-zscc-iam-aws/main.tf]: Done - default is false

File: modules/terraform-zscc-iam-aws/main.tf (hunk 1 above)
Comment (jrguzik) [reply modules/terraform-zscc-iam-aws/main.tf]: Done - default is false

**Comment by jrguzik** [comment]
Done - default is false

File: modules/terraform-zscc-iam-aws/main.tf (hunk 2 above)
Comment (jrguzik) [reply modules/terraform-zscc-iam-aws/main.tf]: Done - default is false

**Comment by jrguzik** [comment]
Added new flag "cloud_tags", with default false.

**Comment by eliassun** [comment]
The new source code looks good. But it conflicts with ASG.

**Review by eliassun — APPROVED** [review]

**Review by jmolnar-zscaler — CHANGES_REQUESTED** [review]
There are permission changes required after talking with the team.

File: modules/terraform-zscc-iam-aws/main.tf [hunk 3]
```diff
@@ -120,6 +120,36 @@ resource "aws_iam_role_policy_attachment" "cc_session_manager_attachment" {
...
+data "aws_iam_policy_document" "cc_tags_policy_document" {
+  version = "2012-10-17"
+  statement {
+    sid    = "CCTags"
+    effect = "Allow"
+    actions = [
+      "sqs:CreateQueue",
+      "sns:Subscribe",
+    ]
+    resources = ["*"]
```
[the remark this answers is not in the document]
Comment (jrguzik) [reply modules/terraform-zscc-iam-aws/main.tf:104]: Only to the iam role and account, the subscribing will be done by janus (and possibly in the future by smedge) When the queue is created a set of different permission will be applied by janus, a future PR (on bitbucket) will show those changes.

**Comment by jmolnar-zscaler** [comment]
closing for https://github.com/zscaler/terraform-aws-cloud-connector-modules/pull/54

**Pull request closed without merging**